src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,218 @@
+                    if (sqlScriptGetters.ContainsKey(type) && sqlObjectTypes.ContainsKey(type))
+                    {
+                        return GetSqlObjectDefinition( 
+                                    sqlScriptGetters[type], 
+                                    tokenText, 
+                                    schemaName, 
+                                    sqlObjectTypes[type]
+                                ); 
+                    }
+                    return null;
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Script a table using SMO
+        /// </summary>
+        /// <param name="tableName">Table name</param>
+        /// <param name="schemaName">Schema name</param>
+        /// <returns>String collection of scripts</returns>
+        internal StringCollection GetTableScripts(string tableName, string schemaName)
+        {
+            return (schemaName != null) ? database.Tables[tableName, schemaName]?.Script()
+                    : database.Tables[tableName].Script();
+        }
+
+        /// <summary>
+        /// Script a view using SMO
+        /// </summary>
+        /// <param name="viewName">View name</param>
+        /// <param name="schemaName">Schema name </param>
+        /// <returns>String collection of scripts</returns>
+        internal StringCollection GetViewScripts(string viewName, string schemaName)
+        {
+            return (schemaName != null) ? database.Views[viewName, schemaName]?.Script()
+                    : database.Views[viewName]?.Script();
+        }
+
+        /// <summary>
+        /// Script a stored procedure using SMO
+        /// </summary>
+        /// <param name="storedProcedureName">Stored Procedure name</param>
+        /// <param name="schemaName">Schema Name</param>
+        /// <returns>String collection of scripts</returns>
+        internal StringCollection GetStoredProcedureScripts(string viewName, string schemaName)
+        {
+            return (schemaName != null) ? database.StoredProcedures[viewName, schemaName]?.Script()
+                    : database.StoredProcedures[viewName]?.Script();
+        }
+
+        /// <summary>
+        /// Script a object using SMO and write to a file.
+        /// </summary>
+        /// <param name="sqlScriptGetter">Function that returns the SMO scripts for an object</param>
+        /// <param name="objectName">SQL object name</param>
+        /// <param name="schemaName">Schema name or null</param>
+        /// <param name="objectType">Type of SQL object</param>
+        /// <returns>Location object representing URI and range of the script file</returns>
+        internal Location[] GetSqlObjectDefinition(
+                ScriptGetter sqlScriptGetter, 
+                string objectName, 
+                string schemaName, 
+                string objectType) 
+        {
+            if (this.connectionInfo.SqlConnection != null)
```

Comment:
This check is here since the database get doesn't check for null. I'll refactor the checks.